### ISSUE TITLE
Some way to quickly see a safe set of features for a set of clients

### ISSUE THREAD

**Issue by osdiab** [hteumeuleu/caniemail#92]
Hi, thanks for the great resource! I wanted to be able to, given a list of clients, get a list of HTML/CSS features that are supported across all of those, for the purpose of creating blacklists of HTML and CSS tags to enforce via linting in my project. There are random blog posts from years ago claiming usable sets of features, but they don't seem exhaustive and the data here seems like it'd be perfect for this kind of use case.

**Comment by hteumeuleu** [comment]
Hi, this is an interesting idea. I feel like this is the sort of things an API would let you do.

**Comment by osdiab** [comment]
Yeah, that would make sense for something like the `caniuse`/`browserify` APIs that are commonly used to decide what features to transpile with tools like babel; though those change a lot more quickly than email support does as far as I know, so even a static file published somewhere on the site (JSON?) would already be very useful (and may itself be the API we're talking about here)

**Comment by hteumeuleu** [comment]
I made data available in JSON at this URL : https://www.caniemail.com/api/data.json
Let me know if you need anything else. Feel free to reopen the issue if needed.

**Comment by nathanv** [comment]
@osdiab did you end up making a linter? What a fantastic idea!

**Comment by osdiab** [comment]
I did not, but if I were to do it, I'd use existing tools to achieve it as opposed to making a linter, like

- if writing emails in JSX: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/forbid-elements.md
- if writing emails with plain HTML: https://html-eslint.org/docs/rules/no-restricted-attrs
- if using CSS stylesheets: https://github.com/stylelint/stylelint/blob/main/lib/rules/property-allowed-list/README.md

basically take the JSON and then plug them into these types of allow/blocklist type rules.

**Comment by kdipippo** [comment]
Adding to this thread that there's an npm cli linter available: https://github.com/shellscape/doiuse-email powered off of the API created in this thread:
- The API file linked earlier: https://www.caniemail.com/api/data.json
- The tool replicates this file locally: https://github.com/shellscape/doiuse-email/blob/main/src/data/can-i-email.json
